src/host/broadcast.jl

```diff
@@ -47,7 +47,15 @@ end
     copyto!(similar(bc, ElType), bc)
 end
 
-@inline function Base.copyto!(dest::BroadcastGPUArray, bc::Broadcasted{Nothing})
+@inline function Base.materialize!(::Style, dest, bc::Broadcasted) where {Style<:AbstractGPUArrayStyle}
+    return _copyto!(dest, instantiate(Broadcasted{Style}(bc.f, bc.args, axes(dest))))
```

Comment:
If this, and the original Base implementation at https://github.com/JuliaLang/julia/blob/17775537232de5b12a38b257e402756b6df55386/base/broadcast.jl#L871, creates a `Broadcasted{<:AbstractGPUArrayStyle}`, why doesn't the `copyto!(::AbstractArray, Broadcasted{<:AbstractGPUArrayStyle})` implementation below suffice?

Reply:
`Base` won't create a `Broadcasted{<:AbstractGPUArrayStyle}`, `bc`'s stype is kept there.
So we need to create `Broadcasted{<:AbstractGPUArrayStyle}` ourself.